File: src/main/java/com/zuhlke/rob/ClipLineListener.java
package com.zuhlke.rob;

import javax.sound.sampled.LineEvent;
import javax.sound.sampled.LineListener;

class ClipLineListener implements LineListener {
    private Object lock;
    private boolean completed = false;

    @Override
    public void update(LineEvent event) {
        LineEvent.Type lineEventType = event.getType();
        if (lineEventType == LineEvent.Type.STOP) {
            completed = true;
            lock.notifyAll();
        }
    }

    public boolean isCompleted() {
        return completed;
    }

    public java.lang.Object playbackLock() {
        Object lock = new Object();
        this.lock = lock;
        return lock;
    }
}

File: src/main/kotlin/com/zuhlke/rob/ClipLineListener.kt
package com.zuhlke.rob

import javax.sound.sampled.LineEvent
import javax.sound.sampled.LineListener

internal class ClipLineListener : LineListener {
    private var lock: PlaybackLock = PlaybackLock()
    var isCompleted = false
        private set

    override fun update(event: LineEvent) {
        val lineEventType = event.type
        if (lineEventType === LineEvent.Type.STOP) {
            isCompleted = true
            lock.complete()
        }
    }

    fun playbackLock(): PlaybackLock {
        return lock
    }
}
